src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -0,0 +1,156 @@
+    /* Returns the resolved value for this setting */                       \
+    /* fallback: user set value --> inherited value --> system set value */ \
+    type name() const                                                       \
+    {                                                                       \
+        const auto val{ _get##name##Impl() };                               \
+        return val ? *val : type{ __VA_ARGS__ };                            \
+    };                                                                      \
+                                                                            \
+    /* Overwrite the user set value */                                      \
+    /* Dispatch event if value changed */                                   \
+    void name(const type& value)                                            \
+    {                                                                       \
+        if (_##name != value)                                               \
+        {                                                                   \
+            _##name = value;                                                \
+        }                                                                   \
+    };                                                                      \
+                                                                            \
+    /* Clear the user set value */                                          \
+    /* Dispatch event if value changed */                                   \
+    void Clear##name()                                                      \
+    {                                                                       \
+        if (Has##name())                                                    \
+        {                                                                   \
+            _##name = std::nullopt;                                         \
+        }                                                                   \
+    };                                                                      \
+                                                                            \
+private:                                                                    \
+    std::optional<type> _##name{ std::nullopt };                            \
+    std::optional<type> _get##name##Impl() const                            \
+    {                                                                       \
+        return _##name ?                                                    \
+                   _##name :                                                \
+                   (_parent ?                                               \
+                        _parent->_get##name##Impl() :                       \
+                        type{ __VA_ARGS__ });                               \

```

Comment:
no since you're doing the `__VA_ARGS__` thing in the real getter on line 61, change this to just return `nullopt`